libs/client/repository/src/lib/content-view/content-view.service.ts

```diff
@@ -0,0 +1,139 @@
+import { Injectable } from '@angular/core';
+import { ContentViewStore } from './content-view.store';
+import { DragonfishNetworkService } from '@dragonfish/client/services';
+import { SelectSnapshot } from '@ngxs-labs/select-snapshot';
+import { UserState } from '@dragonfish/client/repository/user';
+import { FrontendUser } from '@dragonfish/shared/models/users';
+import { ContentKind, SectionInfo } from '@dragonfish/shared/models/content';
+import { of, zip } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { RatingOption } from '@dragonfish/shared/models/reading-history';
+import { ContentViewQuery } from './content-view.query';
+import { AlertsService } from '@dragonfish/client/alerts';
+
+@Injectable({ providedIn: 'root' })
+export class ContentViewService {
+    @SelectSnapshot(UserState.currUser) private currUser: FrontendUser;
+
+    constructor(
+        private contentView: ContentViewStore,
+        private viewQuery: ContentViewQuery,
+        private network: DragonfishNetworkService,
+        private alerts: AlertsService,
+    ) {}
+
+    public fetchContent(contentId: string, kind: ContentKind) {
```

Comment:
Can we get a type annotation on the return value here? It's tough to tell what it's returning without reading the code.